packages/core/src/TextArea/TextArea.test.tsx

```diff
@@ -1,80 +1,85 @@
 import React from "react";
-import { render } from "@testing-library/react";
-import jssSerializer from 'jss-snapshot-serializer';
+import { render, fireEvent } from "@testing-library/react";
+import jssSerializer from "jss-snapshot-serializer";
 expect.addSnapshotSerializer(jssSerializer);
 import ThemeProvider from "../ThemeProvider";
+import { TextAreaProps } from "../types";
 import TextArea from ".";
+import { Formik, Form, Field } from "formik";
 
-const handleChange = jest.fn();
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
+interface OptionalProps extends TextAreaProps {
+  onSubmit?: jest.Mock;
+}
 
-const initialValue =
-  "This is an excerpt, it is a pretty cool excerpt. It contains text.";
-
-test("TextArea renders without crashing, matches snapshot", () => {
-  const { container } = render(
+const TextAreaFormik = (
+  optionalProps?: Partial<OptionalProps>
+): JSX.Element => {
+  const handleSubmit = async (values) => {
+    await sleep(500);
+    optionalProps.onSubmit(values);
+  };
+  return (
     <ThemeProvider>
-      <TextArea
-        id="excerpt"
-        name="excerpt"
-        handleChange={handleChange}
-        field={{
-          name: "excerpt",
-          onChange: jest.fn(),
-          value: initialValue,
-          onBlur: jest.fn(),
-        }}
-        form={{}}
-      />
+      <div>
+        <Formik
+          initialValues={{ textArea: "initial value" }}
+          onSubmit={handleSubmit}
+        >
+          <Form>
+            <Field
+              id="textArea"
+              name="textArea"
+              data-testid
+              component={TextArea}
+              value="initial value"
+              {...optionalProps}
+            />
+          </Form>
+        </Formik>
+      </div>
     </ThemeProvider>
   );
+};
 
+test("TextArea renders without crashing, matches snapshot", () => {
+  const { container } = render(<TextAreaFormik />);
   expect(container).toMatchSnapshot();
 });
 
-// test("Props are passed and rendered correctly", () => {
-//   const { getByTestId } = render(
-//     <ThemeProvider>
-//       <TextArea
-//         id="excerpt"
-//         name="excerpt"
-//         value={initialValue}
-//         handleChange={handleChange}
-//         innerClassName="custom-class"
-//         innerStyle={{ color: "red" }}
-//         style={{ color: "blue" }}
-//         className="another-custom-class"
-//       />
-//     </ThemeProvider>
-//   );
+test("TextArea with label renders without crashing, matches snapshot", () => {
```

Comment:
Along with the snapshot, we should also verify that the value of the label is passed correctly

Reply:
Same ^^